### PR TITLE
Improve --help output + make CI prevent copies of --help output from going out-of-sync

### PR DESCRIPTION
**Pull request by hartwork**
@d-torrance what do you think?

### CODE REVIEW

File: extract-usage-from-readme.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with git-big-picture.  If not, see <http://www.gnu.org/licenses/>.
+
+set -e
+set -o pipefail
+
+fgrep -A55 usage: README.rst | sed 's,^    ,,'
```
Comment (d-torrance) [inline extract-usage-from-readme.sh:22]: Hardcoding 55 as the length of the help output seems hard to maintain.  Perhaps you could do something like this?

```
awk '/usage:/ { print; flag = 1; next } /Usage Examples/ { flag = 0 } flag { print }' README.rst
```

Comment (hartwork) [reply extract-usage-from-readme.sh]: I agree that this approach would probably require fewer updates to file `extract-usage-from-readme.sh` in the future. I have applied that idea now, could you have another look?

Comment (d-torrance) [reply extract-usage-from-readme.sh]: Looks good!

Comment (hartwork) [reply extract-usage-from-readme.sh]: Okay great! Let's see how happy @esc is with the latest version.

**Comment by hartwork** [comment]
I have turned `extract-usage-from-readme.sh` into `rewrite-help-output-in-readme.sh` now so that whoever cħanges `--help` output next, can run a script to sync the readme and will not need to copy paste text around.

**Comment by hartwork** [comment]
@esc I have add documentation of bifurcation commits on top for review here now (for #84)

**Review by esc — APPROVED** [review]
The stuff about bifurcation commits looks good and I like the wording. It's not something in everyones git vocabulary, but I never found a better term for this kind of construct. Thank you for implementing it. I don't know much about manpages, so I can't comment much on that except for: LGTM! 👍

**Pull request merged**